lib/quby/questionnaires/entities/charting/chart.rb

```diff
@@ -10,9 +10,20 @@ class Chart
           attribute :key,                        Symbol
           attribute :title,                      String
           attribute :plottables,                 Array
-          # if categories are defined, plottable values should correspond to values from this array
-          # and the graph will  be plotted with matching y axis categories
+          # if y_categories are defined, plottable values should correspond to values from this array
+          # and the graph will be plotted with corresponding y axis categories
+          # example (icl_r): ["Zeer laag", "Laag", "Gemiddeld", "Hoog", "Zeer Hoog"] (caution, capitalization oddity)
+          # NB: only implemented for bar charts on the roqua side
           attribute :y_categories,               Array
+          # if y_range_categories are defined, plottable values should fall in the ranges that compose the keys of this
+          # hash. The chart will label these ranges of y_values with the corresponding value in the hash on the y axis.
+          # example:
+          # {(0..30) => "Zeer laag", (30..40) => "Laag", (40..60) => "Gemiddeld",
+          # (60..70) => "Hoog", (70..100) => "Zeer hoog"}
```

Comment:
Dit is geen primitive obsession, een hash/map is letterlijk een mapping van domein naar bereik en y_range_categories zijn letterlijk een mapping van domein naar bereik. Op dit moment is het nog niet nodig om op deze hash access op basis van key te hebben die netjes in de ranges zoekt, dus daar een object voor maken is overbodig.

Zoals het nu werkt zijn niet aansluitende ranges niet te gebruiken omdat de plotbands wisselende kleuren moeten hebben. Niet aansluitende ranges zijn niet nuttig omdat je dan een stuk y as hebt zonder interpretatie in je grafiek. 

Ik snap niet hoe je een paar regels setter code 'houtje touwtje en moeilijk aanpasbaar' vindt. Het idee om interpratiecategorieen in scores en grafieken te generaliseren is mooi, maar de huidige implementatie staat dat niet in de weg. Zodra we de wijziging hebben doorgevoerd om een schema te definieren voor scores hebben we daarmee ook een plek om dit soort metadata over scores kwijt te kunnen. Als je een set scores plot die dezelfde categories definieren kan je daarmee bijvoorbeeld automatisch de y_categories of y_range_categories voor de plot instellen en kan deze setter als een override dienen.